### PR TITLE
test: add disk statefulset e2e-test

### LINKED ISSUE #238: add statefulset e2e test cases
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like in detail**

Add following e2e cases which is more similar to real e2e scenarios:
 - run a statefulset which has one pod with disk mount
 - delete the only pod and new pod will be scheduled on the same node
 - wait for pod status as `Running` 
 - if it's not in `Running` state for about 10min, then test fail, otherwise succeed

Config examples:
 - deployment example:
https://github.com/andyzhangx/demo/blob/master/linux/azuredisk/attach-stress-test/deployment/deployment-azuredisk1.yaml

 - statefulset example:
https://github.com/andyzhangx/demo/blob/master/linux/azuredisk/attach-stress-test/statefulset/statefulset-azuredisk1.yaml

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Sakuralbj**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #238 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: test/e2e/testsuites/specs.go [hunk 1]
```diff
@@ -162,6 +162,29 @@ func (pod *PodDetails) SetupDeployment(client clientset.Interface, namespace *v1
...
+	createdStorageClass := tsc.Create()
+	cleanupFuncs = append(cleanupFuncs, tsc.Cleanup)
+	ginkgo.By("setting up the PVC")
+	tpvc := NewTestPersistentVolumeClaim(client, namespace, volume.ClaimSize, volume.VolumeMode, &createdStorageClass)
+	storageClassName := ""
+	if tpvc.storageClass != nil {
+		storageClassName = tpvc.storageClass.Name
+	}
+	tpvc.requestedPersistentVolumeClaim = generateStatefulSetPVC(tpvc.namespace.Name, storageClassName, tpvc.claimSize, tpvc.volumeMode, tpvc.dataSource)
+	//cleanupFuncs = append(cleanupFuncs, tpvc.Cleanup)
```
Comment (andyzhangx) [inline test/e2e/testsuites/specs.go:180]: remove this?

File: test/e2e/testsuites/testsuites.go [hunk 2]
```diff
@@ -504,6 +528,149 @@ func (t *TestDeployment) Logs() ([]byte, error) {
...
+func (t *TestStatefulset) DeletePodAndWait() {
+	e2elog.Logf("Deleting pod %q in namespace %q", t.podName, t.namespace.Name)
+	err := t.client.CoreV1().Pods(t.namespace.Name).Delete(context.TODO(), t.podName, metav1.DeleteOptions{})
+	if err != nil {
+		if !apierrs.IsNotFound(err) {
+			framework.ExpectNoError(fmt.Errorf("pod %q Delete API error: %v", t.podName, err))
+		}
+		return
+	}
+	time.Sleep(40 * time.Second)
```
Comment (andyzhangx) [inline test/e2e/testsuites/testsuites.go:639]: why sleep here?

File: test/e2e/testsuites/specs.go (hunk 1 above)
Comment (Sakuralbj) [reply test/e2e/testsuites/specs.go]: Ok.I remove it.

File: test/e2e/testsuites/testsuites.go (hunk 2 above)
Comment (Sakuralbj) [reply test/e2e/testsuites/testsuites.go]: After pod is deleted, waitForPodready will  ensure the new pod is ready. If don't sleep here,waitForPodready will be passed before old pod is deleted.

Comment (andyzhangx) [reply test/e2e/testsuites/testsuites.go]: ok, pls add a comment before `time.Sleep`.

Comment (andyzhangx) [reply test/e2e/testsuites/testsuites.go]: so statefulset is different from deployment, in deployment, it's not necessary to wait

Comment (Sakuralbj) [reply test/e2e/testsuites/testsuites.go]: Actually,in deployment also use wait,but it use funcition like this.
```
err = e2epod.WaitForPodNoLongerRunningInNamespace(t.client, t.podName, t.namespace.Name)
```
But in statefulset situation, If I use this function,the deleted pod will never be NoLongerRunning State as the new pod has the same name and namespace with old one. So after the new pod created,the test will timeout here. And I choose to use sleep to wait.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**